logp/config.go

```diff
@@ -71,15 +71,19 @@ const (
 // Beat is supposed to be run within.
 func DefaultConfig(environment Environment) Config {
 	toFiles := true
+	toStderr := false
 
-	// If running in a container environment, don't write to files by default.
-	if environment == ContainerEnvironment {
+	// For container and systemd environments, we don't write to files by default.
+	switch environment {
+	case ContainerEnvironment, SystemdEnvironment:
```

Comment:
The fact that we have this logic in two different places, bugs me... Because the FF for 8.15.3 was Tue, I'd get this PR in as it is and backport it if we still have time to make into the release. Then create an issue to re-visit it at some point in the future.

I believe that's the correct function to define the default behaviour, [`createLogOutput`](https://github.com/elastic/elastic-agent-libs/blob/196dace11604a4403f4466812a150ff993c0261c/logp/core.go#L250-L272) should not need the environment switch.